src/Boot/src/Injector/EnumInjector.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spiral\Boot\Injector;
+
+use Spiral\Attributes\AttributeReader;
+use Spiral\Core\Container;
+use Spiral\Core\Container\InjectorInterface;
+use Spiral\Core\Exception\Container\ContainerException;
+
+/**
+ * @internal
+ */
+final class EnumInjector implements InjectorInterface
+{
+    public function __construct(
+        private readonly Container $container,
+        private readonly AttributeReader $reader
+    ) {
+    }
+
+    public function createInjection(\ReflectionClass $class, string $context = null): object
+    {
+        $attribute = $this->reader->firstClassMetadata($class, ProvideFrom::class);
+        if (!$attribute) {
+            throw new ContainerException(
+                \sprintf(
+                    'Class `%s` should contain `%s` attribute with defined detector method.',
+                    $class->getName(),
+                    ProvideFrom::class
+                )
+            );
+        }
+
+        if (!$class->isEnum()) {
+            throw new ContainerException(
+                \sprintf(
+                    'Class `%s` should be an enum.',
+                    $class->getName()
+                )
+            );
+        }
+
+        $object = $this->container->invoke(
+            $class->getMethod($attribute->method)->getClosure()
+        );
```

Comment:
Check here please that `$attribute->method` is valid `callable`
Or you can wrap these lines with a `try/catch` block